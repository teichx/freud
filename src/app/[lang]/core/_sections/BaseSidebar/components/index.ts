export * from './ButtonToggleSidebar';
export * from './SidebarButton';
export * from './SidebarWrapper';
