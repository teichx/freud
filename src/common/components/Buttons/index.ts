export * from './variant';
export * from './types';

import {
  BackButton,
  CancelButton,
  LinkButton,
  SaveButton,
  Microphone,
} from './variant';

export const Buttons = {
  Back: BackButton,
  Cancel: CancelButton,
  Link: LinkButton,
  Save: SaveButton,
  Microphone: Microphone,
};
