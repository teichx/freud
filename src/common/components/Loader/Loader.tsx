import { FC } from 'react';

import { CircularProgress } from '@chakra-ui/react';

import { StyledLoader } from './style';
import { LoaderProps } from './types';

export const Loader: FC<LoaderProps> = ({
  isLoading,
  color = 'book.desertSun.500',
}) => (
  <StyledLoader data-is-loading={isLoading}>
    <CircularProgress isIndeterminate color={color} />
  </StyledLoader>
);
