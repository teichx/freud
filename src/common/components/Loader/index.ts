export * from './Loader';
export * from './style';
export * from './types';
