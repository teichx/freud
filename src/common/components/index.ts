export * from './Avatar';
export * from './Breadcrumb';
export * from './Buttons';
export * from './DataModal';
export * from './DataModal';
export * from './DataTable';
export * from './Display';
export * from './Form';
export * from './Input';
export * from './GoogleLogin';
export * from './InfiniteScroll';
export * from './Loader';
export * from './Logo';
export * from './Meta';
export * from './ReadOnlyText';
export * from './Section';
export * from './Select';
export * from './SkeletonContextLoader';
export * from './TextHighlight';
export * from './ToggleTheme';
export * from './TooltipComponent';
export * from './TooltipDate';
