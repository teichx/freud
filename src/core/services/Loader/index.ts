export * from './LoaderConstants';
export * from './LoaderReducer';
export * from './types';
export * from './useLoader';
