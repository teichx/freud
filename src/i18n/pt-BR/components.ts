export const components = {
  buttons: {
    microphone: {
      toDisable: 'Desligar microfone',
      toEnable: 'Ativar microfone',
      helperText: {
        editOnlyNotListening:
          'O campo não pode ser editado enquanto o microfone estiver ligado',
      },
    },
    login: 'Fazer login',
    access: 'Entrar como {name}',
    logout: 'Deslogar',
    back: 'Voltar',
    cancel: 'Cancelar',
    save: 'Salvar',
  },
  toggleTheme: {
    label: 'Usar tema {appearance}',
    light: 'claro',
    dark: 'escuro',
  },
  form: {
    disabledByPristine: 'Não há alterações a serem salvas',
    disabledByHasValidationErrors: 'Há dados inválidos no formulário',
  },
  dataTable: {
    noData: 'Sem dados por aqui',
    loading: 'Carregando linhas',
    pagination: 'Página {page} de {pageCount}',
    rowsPerPage: 'Linhas por página:',
    pageHint: {
      first: 'Primeira página',
      next: 'Próxima página',
      before: 'Página anterior',
      last: 'Última página',
    },
  },
  select: {
    empty: {
      unfilled: 'Sem mais items',
      filled: 'Não há outros resultados para "{inputValue}"',
    },
    placeholder: {
      unique: 'Selecione uma opção',
      multiple: 'Selecione as opções',
    },
  },
  infiniteScroll: {
    endOfList: 'Sem mais items',
  },
  sidebar: {
    open: 'Abrir barra lateral',
    close: 'Fechar barra lateral',
  },
} as const;
