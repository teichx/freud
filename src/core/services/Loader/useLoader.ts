'use client';
import { useCallback, useState } from 'react';

import { useAppDispatch, useAppSelector } from '~/common/reducer';

import { LoaderSlice } from './LoaderReducer';
import { LoaderType, UseLoader, useLoaderResultProps } from './types';

export const useLoader: UseLoader = (loader, ...identifiersParam) => {
  const [identifiers] = useState([loader, ...identifiersParam]);
  const loaders = useAppSelector((x) => x.loader);
  const isLoading = Object.entries(loaders)
    .filter(([key]) => identifiers.includes(key as LoaderType))
    .some(([, x]) => x.isLoading);
  const dispatch = useAppDispatch();

  const setIsLoading = useCallback<useLoaderResultProps['setIsLoading']>(
    (isLoadingParam) =>
      dispatch(
        LoaderSlice.actions.setIsLoading({
          identifiers,
          isLoading: isLoadingParam,
        })
      ),
    [dispatch, identifiers]
  );

  const startLoading = useCallback<useLoaderResultProps['startLoading']>(
    () =>
      dispatch(
        LoaderSlice.actions.setIsLoading({ identifiers, isLoading: true })
      ),
    [dispatch, identifiers]
  );

  const endLoading = useCallback<useLoaderResultProps['endLoading']>(
    () =>
      dispatch(
        LoaderSlice.actions.setIsLoading({ identifiers, isLoading: false })
      ),
    [dispatch, identifiers]
  );

  return {
    isLoading,
    setIsLoading,
    startLoading,
    endLoading,
  };
};
