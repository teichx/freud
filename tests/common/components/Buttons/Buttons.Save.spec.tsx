jest.mock('~/core/services/Loader/useLoader');
import { render } from 'test-utils';

import { Buttons } from '~/common/components/Buttons';
import * as services from '~/core/services/Loader/useLoader';

describe('component.Buttons', () => {
  describe('buttons.Save', () => {
    beforeEach(() => {
      jest.spyOn(services, 'useLoader').mockReturnValue({
        isLoading: false,
        setIsLoading: jest.fn(),
        startLoading: jest.fn(),
        endLoading: jest.fn(),
      });
    });

    it('default', () => {
      expect.assertions(2);

      const { asFragment, getByText } = render(<Buttons.Save />);

      expect(asFragment()).toMatchSnapshot();
      expect(getByText('words.save')).toBeDefined();
    });

    it('with text', () => {
      expect.assertions(1);
      const text = '<current-text>';

      const { getByText } = render(<Buttons.Save text={text} />);

      expect(getByText(text)).toBeDefined();
    });

    it('with children', () => {
      expect.assertions(1);

      const text = '<current-children>';

      const { getByText } = render(<Buttons.Save>{text}</Buttons.Save>);

      expect(getByText(text)).toBeDefined();
    });
  });

  describe('buttons.Save when isLoading', () => {
    it('default when isLoading is not equal not loading', () => {
      expect.assertions(1);
      jest.spyOn(services, 'useLoader').mockReturnValue({
        isLoading: false,
        setIsLoading: jest.fn(),
        startLoading: jest.fn(),
        endLoading: jest.fn(),
      });

      const { asFragment: firstAsFragment } = render(<Buttons.Save />);

      jest.spyOn(services, 'useLoader').mockReturnValue({
        isLoading: true,
        setIsLoading: jest.fn(),
        startLoading: jest.fn(),
        endLoading: jest.fn(),
      });

      const { asFragment: secondAsFragment } = render(<Buttons.Save />);

      expect(firstAsFragment()).not.toStrictEqual(secondAsFragment());
    });

    it('default when isLoading', () => {
      expect.assertions(2);
      jest.spyOn(services, 'useLoader').mockReturnValue({
        isLoading: true,
        setIsLoading: jest.fn(),
        startLoading: jest.fn(),
        endLoading: jest.fn(),
      });

      const { asFragment, getByText } = render(<Buttons.Save />);

      expect(asFragment()).toMatchSnapshot();
      expect(getByText('words.save')).toBeDefined();
    });
  });
});
