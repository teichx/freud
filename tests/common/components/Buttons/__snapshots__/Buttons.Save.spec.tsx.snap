// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.Buttons buttons.Save default 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-pecn96"
    type="submit"
  >
    <span
      class="chakra-button__icon css-1wh2kri"
    >
      <svg
        aria-hidden="true"
        fill="none"
        focusable="false"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h11l5 5v11a2 2 0 0 1-2 2z"
        />
        <polyline
          points="17 21 17 13 7 13 7 21"
        />
        <polyline
          points="7 3 7 8 15 8"
        />
      </svg>
    </span>
    translations.words.save
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`component.Buttons buttons.Save when isLoading default when isLoading 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-pecn96"
    data-loading=""
    disabled=""
    type="submit"
  >
    <div
      class="chakra-button__spinner chakra-button__spinner--start css-1aamklj"
    >
      <div
        class="chakra-spinner css-1fc7edv"
      >
        <span
          class="css-8b45rq"
        >
          Loading...
        </span>
      </div>
    </div>
    <span
      class="css-q7lffx"
    >
      <span
        class="chakra-button__icon css-1wh2kri"
      >
        <svg
          aria-hidden="true"
          fill="none"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h11l5 5v11a2 2 0 0 1-2 2z"
          />
          <polyline
            points="17 21 17 13 7 13 7 21"
          />
          <polyline
            points="7 3 7 8 15 8"
          />
        </svg>
      </span>
      translations.words.save
    </span>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
